integration/nestjs-simple/hero.ts

```diff
@@ -27,6 +30,58 @@ export interface Villain {
 
 export const HERO_PACKAGE_NAME = "hero";
 
+const wrapStruct = (value: any, nested = false): any => {
+  const valueType = typeof value;
+  const primitiveValueTypes = { number: "numberValue", string: "stringValue", boolean: "boolValue" };
+  if (Object.keys(primitiveValueTypes).includes(valueType)) {
+    return Value.wrap(value);
+  }
+  if (Array.isArray(value)) {
+    return { listValue: { values: value.map((item) => wrapStruct(item)) } };
+  }
+  if (valueType === "object") {
+    const res = nested ? { structValue: { fields: {} as any } } : { fields: {} as any };
+    Object.keys(value).forEach((field) => {
+      if (nested) {
+        res.structValue!.fields[field] = wrapStruct(value[field], true);
+      } else {
+        res.fields![field] = wrapStruct(value[field], true);
```

Comment:
Question: Should this `true` be a false? Just thinking that we are in the `nested = false` branch of the `it`, so should we pass along `false` as well?

Reply:
The only case we actually need for `nested` to be `false` is the base object, we should only get here initially and from here it should be always `true`

Reply:
Gotcha, thanks!